### PR TITLE
Remove useless count in ValidatorListener

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Bridge/Symfony/Validator/EventListener/ValidateListener.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public function onKernelView(GetResponseForControllerResultEvent $event)
         }
 
         $violations = $this->validator->validate($data, null, $validationGroups);
-        if (0 !== count($violations)) {
+        if ($violations) {
```
Comment (soyuka) [inline src/Bridge/Symfony/Validator/EventListener/ValidateListener.php]: This evaluates to `true` even if there are no violations. `count` isn't that useless here.

**Comment by teohhanhui** [comment]
I really think we should prefer to be explicit, instead of trying to micro-optimize (we should do that where it matters).

**Comment by soyuka** [comment]
I think that @dunglas just saw that while trying to fix the segfault and therefore changed it (in the train with 3G). No big deal, closing this one.

**Comment by teohhanhui** [comment]
Someone needs to send a debug core dump to the PHP team. I'm running into segfault too with phpdbg + PHPUnit in our project. :cry:

**Comment by soyuka** [comment]
https://github.com/api-platform/core/pull/979 ^^

I'm having segfaults because of either the ulimit either the memory limit when covering the Configuration test case.

**Pull request closed without merging**